cpp/src/sort/is_sorted.cu

```diff
@@ -36,10 +36,12 @@ auto is_sorted(
 {
     cudaStream_t stream = 0;
     auto in_d = table_device_view::create(in);
+    rmm::device_vector<order> d_column_order(column_order);
+    rmm::device_vector<null_order> d_null_precedence(null_precedence);
```

Comment:
Unsure if this should somehow be conditioned based on the template because this is an unnecessary host --> device if there's no nulls. Would appreciate guidance here.

Reply:
excerpt from `sorted_order`, If looks like `d_null_precedence` isn't passed at all in the no-nulls case:

```
  if (has_nulls(input)) {
    rmm::device_vector<null_order> d_null_precedence(null_precedence);
    auto comparator = row_lexicographic_comparator<true>(
        *device_table, *device_table,
        d_column_order.data().get(),
        d_null_precedence.data().get());
    thrust::sort(rmm::exec_policy(stream)->on(stream),
                 mutable_indices_view.begin<int32_t>(),
                 mutable_indices_view.end<int32_t>(), comparator);

  } else {
    auto comparator = row_lexicographic_comparator<false>(
        *device_table, *device_table,
        d_column_order.data().get());
    thrust::sort(rmm::exec_policy(stream)->on(stream),
                 mutable_indices_view.begin<int32_t>(),
                 mutable_indices_view.end<int32_t>(), comparator);
  }
```

May be able to reduce the amount of code, though?
```
rmm::device_vector<null_order> d_null_precedence;

if (has_nulls) {
    d_null_precedence = null_precedence;
}
```
and
```
auto ineq_op = ...(
    ... ,
   has_nulls ? d_null_precedence.data().get() : nullptr);

```
```

Reply:
You're going to make @codereport sad with this "Initialize then Modify".